src/playground-ide.ts

```diff
@@ -147,17 +201,67 @@ export class PlaygroundIde extends LitElement {
       >
       </playground-file-editor>
 
-      <playground-preview
-        part="preview"
-        exportparts="preview-toolbar,
-                     preview-location,
-                     preview-reload-button,
-                     preview-loading-indicator"
-        location="Result"
-        .project=${projectId}
-      ></playground-preview>
+      <div id="rhs">
+        ${this.resizable
+          ? html`<div
+                id="resizeBar"
+                @mousedown=${this.onResizeBarMousedown}
+              ></div>
+              <div id="resizeOverlay"></div>`
+          : nothing}
+
+        <playground-preview
+          part="preview"
+          exportparts="preview-toolbar,
+                       preview-location,
+                       preview-reload-button,
+                       preview-loading-indicator"
+          location="Result"
+          .project=${projectId}
+        ></playground-preview>
+      </div>
     `;
   }
+
+  async update(changedProperties: PropertyValues<this>) {
+    super.update(changedProperties);
+    if (changedProperties.has('resizable') && this.resizable === false) {
+      // Note we set this property on the RHS element instead of the host so
+      // that when "resizable" is toggled, we don't reset a host value that the
+      // user might have set.
+      this._rhs.style.removeProperty('--playground-preview-width');
+    }
+  }
+
+  private onResizeBarMousedown() {
+    const rhsStyle = this._rhs.style;
+    const overlay = this._resizeOverlay;
+    overlay.classList.add('resizing');
+
+    const {left: hostLeft, right: hostRight} = this.getBoundingClientRect();
+    const hostWidth = hostRight - hostLeft;
+    const rhsMinWidth = 100;
+    const rhsMaxWidth = hostWidth - 100;
+
+    const onMousemove = (event: MouseEvent) => {
+      const rhsWidth = Math.min(
+        rhsMaxWidth,
+        Math.max(rhsMinWidth, hostRight - event.clientX)
+      );
+      const percent = (rhsWidth / hostWidth) * 100;
+      rhsStyle.setProperty('--playground-preview-width', `${percent}%`);
+    };
+    overlay.addEventListener('mousemove', onMousemove);
+
+    const stopDragging = () => {
+      overlay.classList.remove('resizing');
+      overlay.removeEventListener('mousemove', onMousemove);
+      overlay.removeEventListener('mouseup', stopDragging);
+      overlay.removeEventListener('mouseout', stopDragging);
+    };
+    overlay.addEventListener('mouseup', stopDragging);
+    overlay.addEventListener('mouseout', stopDragging);
```

Comment:
why not use pointerevents with capture to prevent this weird use case? It's now supported on safari (though unfortunately not yet safari -2 versions)

Reply:
Because I'd never heard of that before. Thanks! Switched, tested in latest Chrome/Firefox/Safari.